app/rtc_compare.py

```diff
@@ -0,0 +1,483 @@
+
+        flag_identical_attrs_structure = \
+            (len(intersection_set_attrs) == len(set_attrs_1) and
+             len(intersection_set_attrs) == len(set_attrs_2))
+
+        # Proceed with checking the values in attributes,
+        # regardless of the agreement of their structure.
+
+        list_flag_identical_attrs = [None] * len(intersection_set_attrs)
+        for id_flag, key_attr in enumerate(intersection_set_attrs):
+            list_flag_identical_attrs[id_flag] = \
+                compare_hdf5_elements(hdf5_in_1,
+                                      hdf5_in_2,
+                                      key_attr,
+                                      is_attr=True,
+                                      id_key=id_flag,
+                                      total_key=len(intersection_set_attrs),
+                                      print_passed_element=False)
+
+        flag_same_dataset = all(list_flag_identical_dataset)
+        flag_same_attributes = all(list_flag_identical_attrs)
+
+        final_result = (flag_identical_dataset_structure and
+                        flag_identical_attrs_structure and
+                        flag_same_dataset and
+                        flag_same_attributes)
+
+        print('\n\nTest completed.')
+
+        if flag_identical_dataset_structure:
+            print('\nDataset structure identical.')
+        else:
+            print('\nDataset structure not identical.')
+            print('In the 1st HDF5, not in the 2nd data:')
+            print('\n'.join(list(set_dataset_1 - set_dataset_2)))
+            print('In the 2st HDF5, not in the 1nd data:')
+            print('\n'.join(list(set_dataset_2 - set_dataset_1)))
+
+        if flag_identical_attrs_structure:
+            print('Attribute structure identical.')
+
+        else:
+            flag_identical_attrs_structure = False
+            print('\nAttribute structure not identical.')
+            print('In the 1st HDF5, not in the 2nd data:')
+            print('\n'.join(list(set_dataset_1 - set_dataset_2)))
+            print('In the 2nd HDF5, not in the 1st data:')
+            print('\n'.join(list(set_dataset_2 - set_dataset_1)))
+
+        if all(list_flag_identical_dataset):
+            print( 'The datasets of the two HDF files are the same within '
+                  f'the threshold of {RTC_S1_PRODUCTS_ERROR_TOLERANCE}')
+        else:
+            print(f'{sum(~np.array(list_flag_identical_dataset))} datasets out of '
+                  f'{len(intersection_set_dataset)} did not pass the test.')
+
+        if all(list_flag_identical_attrs):
+            print( 'The attributes of the two HDF files are the same within '
+                  f'the threshold of {RTC_S1_PRODUCTS_ERROR_TOLERANCE}\n')
+        else:
+            print(f'{sum(~np.array(list_flag_identical_attrs))} attributes out of '
+                  f'{len(intersection_set_attrs)} did not pass the test.\n')
```

Comment:
These are the final results. Maybe you can add the nice green/red PASSED/FAILED messages to these prints. Just to highlight the results.

Reply:
Good idea. I've put the colorized PASSED / FAILED message at the beginning of the prints.